pkg/generator/github/github.go

```diff
@@ -80,8 +83,21 @@ func (g *Generator) generate(
 	if err != nil {
 		return nil, fmt.Errorf("error creating request: %w", err)
 	}
+
+	body := make(map[string]interface{})
+	if gh.Permissions != nil {
+		body["permissions"] = gh.Permissions
+	}
+	if len(gh.Repositories) > 0 {
+		body["repositories"] = gh.Repositories
+	}
+	bodyBytes, err := json.Marshal(body)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
 	// Github api expects POST request
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, gh.URL, http.NoBody)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, gh.URL, bytes.NewReader(bodyBytes))
```

Comment:
This is slightly different as opposed to not sending a body at all. I would say, make the body nil and only created one if the things are set.

so something like:

```go
body := http.NoBody
...
if bla
body, err = bytes.NewReader(bodyBytes)
...


...
req, err := http.NewRequestWithContext(ctx, http.MethodPost, gh.URL, body)
```